src/app/InteractionModelEngine.h

```diff
@@ -124,7 +125,10 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate
      */
     uint16_t GetReadClientArrayIndex(const ReadClient * const apReadClient) const;
 
+    reporting::ReportingEngine * GetReportingEngine(void) { return &mReportingEngine; }
```

Comment:
do not use `(void)`. Use `()` instead.

Why does this return a pointer instead of a reference?

Reply:
sure, removed, also file one ticket to remove all (void) in IM.
agree to return a reference, thanks for pointing out it.

Reply:
https://github.com/project-chip/connectedhomeip/issues/5737